runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/grpcclient/GrpcWindmillServer.java

```diff
@@ -238,34 +237,29 @@ public synchronized boolean isReady() {
   private synchronized void initializeLocalHost(int port) {
     this.logEveryNStreamFailures = 1;
     this.maxBackoff = Duration.millis(500);
-    this.unaryDeadlineSeconds = 10; // For local testing use short deadlines.
     Channel channel = localhostChannel(port);
     if (options.isEnableStreamingEngine()) {
       this.stubList.add(CloudWindmillServiceV1Alpha1Grpc.newStub(channel));
-      this.syncStubList.add(CloudWindmillServiceV1Alpha1Grpc.newBlockingStub(channel));
     } else {
-      this.syncApplianceStub = WindmillApplianceGrpc.newBlockingStub(channel);
+      this.syncApplianceStub =
+          WindmillApplianceGrpc.newBlockingStub(channel).withDeadlineAfter(10, TimeUnit.SECONDS);
```

Comment:
I remembered that I think this doesn't work as expected. The deadline is translated to an absolute timestamp here and thus all calls made with this stub will timeout after 10 seconds from when this is called (instead of a timeout on each call).

Could look into adding a test about that or at least add a comment for next time on why it is done for each call.

See https://github.com/grpc/grpc-java/issues/1495

Reply:
There may be a way to bind a call interceptor to set the deadline each time and thus set it up once here still.

Reply:
grpc/grpc-java/issues/4305 also found this which says that the stubs are meant to be forked and not cached. i will update

Reply:
done. added an interceptor and documentation thanks for pointing this out